onnx/defs/nn/defs.cc

```diff
@@ -2832,4 +2832,101 @@ ONNX_OPERATOR_SET_SCHEMA(
               schema.BuildFunction(functionProto);
               return true;
             }));
+
+static const char* RotaryEmbedding_ver23_doc = R"DOC(
+RotaryEmbedding is the implementation of rotary positional embeddings (RoPE) based on the paper https://arxiv.org/pdf/2104.09864.
+The key advantage of RoPE is that it allows the model to understand both the absolute position of a token and the relative distances
+between tokens. This is achieved through a rotational mechanism where the extent of rotation is computed based on the token's absolute position (position_ids).
+
+The rotational mechanism is defined by sine and cosine functions that are used to represent the rotation angles.
+For each token in the sequence, its positional embedding is computed by rotating its embedding vector. This is done by splitting the
+embedding vector into two halves and applying the rotation matrix to each half of the embedding vector. The rotation matrix is
+parameterized by the token's position in the sequence. The rotated halves of the embedding vector are concatenated to form the final positional
+embedding for each token. The rotated positional embeddings are used in the self-attention mechanism. The rotation ensures that the model
+captures both absolute and relative positional information.
+
+Rotary embeddings are defined using the following algorithm:
+
+    def rotate_half(x):
```

Comment:
Please align this definition with the function definition below and the implementation in the test-file generation as much as possible. It helps the reader. (We should have minimal differences, only those necessary due to differences between numpy and onnx ops.)